src/parser/demonhunter/havoc/modules/talents/Demonic.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import Events from 'parser/core/Events';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SpellLink from 'common/SpellLink';
+
+/**
+ * Example Report: https://www.warcraftlogs.com/reports/23dHWCrT18qhaJbz/#fight=1&source=16
+ */
+
+const META_BUFF_DURATION_EYEBEAM = 10000;
+
+class Demonic extends Analyzer{
+
+  eyeBeamCasts = 0;
+  goodDeathSweep = 0;
+  eyeBeamTimeStamp = undefined;
+  counter = undefined;
+  badCasts = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.DEMONIC_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EYE_BEAM), this.getTimeStamp);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.DEATH_SWEEP), this.countCasts);
+  }
+
+  getTimeStamp(event) {
+    const hasMetaBuff = this.selectedCombatant.hasBuff(SPELLS.METAMORPHOSIS_HAVOC_BUFF.id, event.timestamp - 1000);
```

Comment:
Why should the player have Metamorphosis 1 second before the event and not at the time of the event?

Reply:
In this case i am checking for eye beam casts that are just for the dps and ignoring eyebeam casts that are used to extend meta time.
If you manually casted meta and then cast eyebeam the meta u get from eyebeam extends the meta you casted, which you might do and ignore if deathsweap is up to be casted since that takes priority. In this case i wanted to check that this wasnt a meta extending cast and if it isnt you should be saving up ur death sweaps to use in that meta window.